### PR TITLE
19044 - Amalgamation unit tests

### PR DESCRIPTION
**Pull request by ketaki-deodhar**
*Issue #:* /bcgov/entity#19044

*Description of changes:*

Added unit tests in:

- AmalgamationBusinessInfo.spec.ts
- AmalgamationInformation.spec.ts
- AmalgamationPeopleRoles.spec.ts
- AmalgamationReviewConfirm.spec.ts
- PeopleAndRoles.spec.ts

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: tests/unit/AmalgamatingBusinesses.spec.ts [hunk 1]
```diff
@@ -73,11 +73,11 @@ describe('Amalgamating Businesses - components and validity', () => {
   })
 
   it('computes havePrimaryBusiness correctly', () => {
-    // verify with no holding businesses
+    // verify with no primary businesses
```
Comment (ketaki-deodhar) [inline tests/unit/AmalgamatingBusinesses.spec.ts:76]: Updated comment based on test

File: tests/unit/AmalgamationInformation.spec.ts [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+import { AmalgamationRegResources, AmalgamationShortResources } from '@/resources/'
 import AmalgamatingBusinesses from '@/components/Amalgamation/AmalgamatingBusinesses.vue'
 import { ExpandableHelp } from '@bcrs-shared-components/expandable-help'
 import ResultingBusinessName from '@/components/Amalgamation/ResultingBusinessName.vue'
 import BusinessTypeHelp from '@/components/Amalgamation/BusinessTypeHelp.vue'
 import { AmalgamationTypes, FilingTypes } from '@bcrs-shared-components/enums'
 
 // Test Case Data
-const amalgamationRegularBusinessInfo = [
+const amalgamationBusinessInfo = [
```
Comment (ketaki-deodhar) [inline tests/unit/AmalgamationInformation.spec.ts:12]: renamed as can be used for regular and short tests

File: tests/unit/AmalgamationReviewConfirm.spec.ts [hunk 3]
```diff
@@ -125,4 +129,90 @@ for (const test of amalgamationRegularBusinessInfo) {
       expect(section.findComponent(StaffPayment).exists()).toBe(true)
     })
   })
+
```
Comment (ketaki-deodhar) [inline tests/unit/AmalgamationReviewConfirm.spec.ts:132]: Added a few tests for Short amalgamation (it is a bit of a different case than other filings)

File: tests/unit/PeopleAndRoles.spec.ts [hunk 4]
```diff
@@ -167,3 +168,57 @@ describe('People And Roles component', () => {
     resetStore()
   })
 })
+
```
Comment (ketaki-deodhar) [inline tests/unit/PeopleAndRoles.spec.ts:171]: Added a few tests for Short amalgamation (it is a bit of a different case than other filings)

File: tests/unit/AmalgamationInformation.spec.ts [hunk 5]
```diff
@@ -1,15 +1,15 @@
 import { shallowWrapperFactory } from '../vitest-wrapper-factory'
 import { AmalgamationInformation } from '@/views'
 // *** FUTURE: add tests for AmalgamationShortResources
```
Comment (severinbeauvais) [inline tests/unit/AmalgamationInformation.spec.ts:3]: Please remove this comment now.

**Review by severinbeauvais — COMMENTED** [review]
What about this one?

https://github.com/bcgov/business-create-ui/blob/964251f344e736290f16717841450cbee4a810ad/tests/unit/AmalgamationShareStructure.spec.ts#L32

**Comment by ketaki-deodhar** [comment]
> What about this one?
> 
> https://github.com/bcgov/business-create-ui/blob/964251f344e736290f16717841450cbee4a810ad/tests/unit/AmalgamationShareStructure.spec.ts#L32

For Short form, Share structure is not a step and is only shown in Review and Confirm step as read only.

**Comment by severinbeauvais** [comment]
> For Short form, Share structure is not a step and is only shown in Review and Confirm step as read only.

Oh, right. Please remove that comment. Thanks!

**Review by jamespaologarcia — APPROVED** [review]
LGTM

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**